R/pandoc.R

```diff
@@ -472,44 +472,77 @@ pandoc_self_contained_html <- function(input, output) {
 }
 
 
-validate_self_contained <- function(mathjax) {
+validate_self_contained <- function(math) {
 
-  if (identical(mathjax, "local"))
-    stop2("Local MathJax isn't compatible with self_contained\n",
-         "(you should set self_contained to FALSE)")
+  if (identical(c(names(math), math[[1L]]), c("mathjax", "local")))
+    stop("Local MathJax isn't compatible with self_contained\n",
+         "(you should set self_contained to FALSE)", call. = FALSE)
 }
 
-pandoc_mathjax_args <- function(mathjax,
-                                template,
-                                self_contained,
-                                files_dir,
-                                output_dir) {
+pandoc_math_args <- function(math,
+                             template,
+                             self_contained,
+                             files_dir,
+                             output_dir) {
+  url <- math[[1L]]
+
+  if (is.null(url)) {
+    return(NULL)
+  }
+
+  engine <- names(math)
+
+  if (is.null(engine)) {
+    if (url %in% c("mathml", "webtex", "gladtex")) {
+      return(paste0("--", math[[1L]]))
+    }
+
+    if (identical(url, "mathjax")) {
+      url <- "default"
+      engine <- "mathjax"
+    } else if (identical(url, "katex")) {
+      url <- "default"
+      engine <- "katex"
+    }
+  }
+
+  if (identical(engine, "webtex")) {
+    return(if (identical(url, "default")) {
+      "--webtex"
+    } else {
+      paste0("--webtex=", url)
+    })
+  }
```

Comment:
This is because `mathml` and `gladtex` does not require URL like `--mathml=url`.
Yet, my implementation lacks the alignment...

https://pandoc.org/MANUAL.html#math-rendering-in-html

I think the following form which you suggest is the better.
In this way, we can simply ignore `url` for `mathml` and `gladtex`

```yaml
output: 
  html_document:
    math: 
       engine: katex
       url: https://cdn.jsdelivr.net/npm/katex@0.12.0/dist"
```

Reply:
This is the supported syntax now
````yaml
output: 
  html_document:
    math: 
       engine: katex
       url: https://cdn.jsdelivr.net/npm/katex@0.12.0/dist"
````